packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,221 @@
+      credential: IVerifiableCredential
+    ) => Promise<VerificationResult>
+  ) {
+    let hasParent = true;
+    let didMatched = false;
+    while (hasParent) {
+      const role = await this.parseRoleFromCredential(credential);
+      const issuers = await this._issuerDefResolver.getIssuerDefinition(role);
+      if (issuers.did && issuers.did.length > 0) {
+        for (let i = 0; i < issuers.did.length; i++) {
+          if (issuers.did[i] == credential.issuer) {
+            didMatched = true;
+            hasParent = false;
+            break;
+          }
+        }
+      } else {
+        const issuerCredential = await this._credentialResolver.getCredential(
+          credential.issuer,
+          role
+        );
+        // @ts-ignore
+        const result = await verifyCredentialProofCallback(issuerCredential);
+        if (result) {
+          // @ts-ignore
+          credential = issuerCredential;
+        } else {
+          throw new Error('Invalid credential');
+        }
+      }
+    }
+  }
+
+  /**
+   * Fetches role form a credential
+   * @param credential
+   * @returns
+   */
+  private async parseRoleFromCredential(credential: IVerifiableCredential) {
+    return credential.credentialSubject.role.namespace;
+  }
+
+  /**
+   * Returns true if the role issuer type is DID
+   * @param role
+   * @returns
+   */
+  private async isRoleIssuerDID(role: string) {
+    const issuers = await this._issuerDefResolver.getIssuerDefinition(role);
+    return issuers.did && issuers.did.length > 0;
+  }
+
+  /**
+   * Verify issued token signature
+   * @param {string} token
+   */
+  async verifyIssuedToken(token: string) {
+    const offChainClaim = jwt.decode(token) as OffChainClaim;
+    const issuerDIDDoc = await this._resolver.read(offChainClaim.iss);
+    const verifier = new ProofVerifier(issuerDIDDoc);
+    if (await verifier.verifyAssertionProof(token)) {
+      return offChainClaim.iss;
+    } else {
+      throw new Error('Invalid Credential');
+    }
+  }
+
+  /**
+   * Verifies issuer's authority to issue credential
+   * @param {string} namespace
+   * @param {string} issuerDID
+   * @returns boolean
+   */
+  async verifyIssuerAuthority(
+    namespace: string,
+    issuerDID: string
+  ): Promise<boolean> {
+    const issuers = await this._issuerDefResolver.getIssuerDefinition(
+      namespace
+    );
+    if (issuers.did && issuers.did.length > 0) {
+      for (let i = 0; i < issuers.did.length; i++) {
+        if (issuers.did[i] == addressOf(issuerDID)) {
+          return true;
+        }
+      }
+    }
+    let offChainClaim;
+    if (issuers.roleName) {
+      offChainClaim = await this._credentialResolver.getCredential(
+        issuerDID,
+        issuers.roleName
+      );
+    }
+    if (!offChainClaim) {
+      return false;
+    } else {
+      const isClaimVerified = await this.verifyIssuedToken(
+        offChainClaim.issuedToken
+      );
+      return typeof isClaimVerified === 'string';
+    }
```

Comment:
```suggestion
    }
    const isClaimVerified = await this.verifyIssuedToken(
      offChainClaim.issuedToken
    );
    return typeof isClaimVerified === 'string';
```
I think we don't need to nest it in `else` statement. Less indentation == more readable code, imo